web/js/function-highlight-addon.js

```diff
@@ -0,0 +1,176 @@
+// CodeMirror, copyright (c) by Marijn Haverbeke and others
```

Comment:
Yeah, it does need to go in third_party, with the original license.  I'm not a lawyer, so I'm  just going to follow Google's rules here, rather than speculate on the ability to change licenses. :)